traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java

```diff
@@ -140,7 +148,57 @@ public void onClick(View v) {
             }
         });
 
+        wishListToggleImageButton = view.findViewById(R.id.imagebutton_wishlist_toggle);
+        wishListToggleImageButton.setSelected(catalogItemDetails.isWishlisted());
+        wishListToggleImageButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                List<Product> products= new ArrayList<>();
+                products.add(catalogItemDetails);
+                if (wishListToggleImageButton.isSelected()) {
+                    WishlistResult immediateResult = Traveler.wishlistRemove(
+                            products, wishlistResult,
+                            CatalogItemDetailsFragment.this);
+                    applyWishlistResult(immediateResult);
+
+                } else {
+                    wishListToggleImageButton.setSelected(true);
+                    Traveler.wishlistAdd(products, CatalogItemDetailsFragment.this);
+                }
+            }
+        });
         return view;
     }
 
+    @Override
+    public void onWishlistAddSuccess(List<CatalogItem> products) {
+        // no-op
```

Comment:
use this for this screen. you dont need the result

You dont even have the results.

Has this been tested?

Reply:
I had it setup so:
1) when the user clicks on the heart
2) immediately flip heart to red, then make call
3) if the call fails, reset heart to grey
3) if the call succeeds, do nothing

so I can change the process to

1) when the user clicks on the heart
2) make the call 
3) if the call succeeds, set heart to red
3) if the fall fails, do nothing
~~~~~~
I can add a check for the results in the full integration test

Reply:
do the following:

1. user clicks
2. make ui changes immediately
3. call the sdk api
4. if there is an error, return item state to original and revert ui